cpp/benchmarks/CMakeLists.txt

```diff
@@ -14,7 +14,12 @@
 
 find_package(Threads REQUIRED)
 
-add_library(cudf_datagen STATIC common/generate_input.cpp)
+add_library(cudf_datagen STATIC common/generate_input.cu)
+set_property(
+  SOURCE common/generate_input.cu
+  APPEND
+  PROPERTY COMPILE_FLAGS "-Xcompiler=-std=gnu++17"
```

Comment:
Adding `CXX_EXTENSIONS ON` does not seem to work!
@robertmaynard can you point what is wrong with following cmake code?
```
set_target_properties(
  cudf_datagen
  PROPERTIES CXX_STANDARD 17
             CXX_STANDARD_REQUIRED ON
             # For std:: support of __int128_t. Can be removed once using cuda::std
             CXX_EXTENSIONS ON
             CUDA_STANDARD 17
             CUDA_STANDARD_REQUIRED ON
)
```

Reply:
`CXX_STANDARD` controls the C++ compiler only it has no impact on the CUDA compiler or the underlying C++ compiler it invokes. The fact that `nvcc` invokes a C++ compiler is an implementation detail and not something required for all CUDA compilers.

Since `nvcc` doesn't expose a highlevel `-std=[gnu]...` set of flags, CMake doesn't map the `CUDA_EXTENSIONS` property to `-Xcompiler=[]` values. Plus as it currently stands CMake doesn't do sufficient `nvcc` inspection to know what compiler `nvcc` is invoking and  what `std` flags it supports ( MSVC, IBM, clang, ... )

Reply:
Restructured the code to cpp, cu files separately. gnu++17 not needed for nvcc.
Thank you @robertmaynard